Library/ViewModels/AddNewCardViewModelTests.swift

```diff
@@ -0,0 +1,209 @@
+    self.paymentDetailsBecomeFirstResponder.assertDidEmitValue()
+    self.vm.inputs.creditCardChanged(cardNumber: "4242 4242 4242 4242",
+                                      expMonth: 11,
+                                      expYear: 99,
+                                      cvc: "123")
+    self.vm.inputs.paymentInfo(valid: true)
+    self.saveButtonIsEnabled.assertValues([true])
+
+    self.vm.inputs.saveButtonTapped()
+    self.activityIndicatorShouldShow.assertValues([true])
+
+    self.vm.inputs.stripeCreated("stripe_deadbeef", stripeID: "stripe_deadbeefID")
+
+    self.scheduler.advance()
+
+    self.addNewCardSuccess.assertValues([Strings.Got_it_your_changes_have_been_saved()])
+    self.activityIndicatorShouldShow.assertValues([true, false])
+  }
+
+  func testAddCardFailure_InvalidToken() {
+    self.vm.inputs.viewDidLoad()
+    self.vm.inputs.cardholderNameChanged("Native Squad")
+    self.vm.inputs.cardholderNameTextFieldReturn()
+    self.paymentDetailsBecomeFirstResponder.assertDidEmitValue()
+
+    self.vm.inputs.creditCardChanged(cardNumber: "4242 4242 4242 4242",
+                                      expMonth: 11,
+                                      expYear: 99,
+                                      cvc: "123")
+    self.vm.inputs.paymentInfo(valid: true)
+    self.saveButtonIsEnabled.assertValues([true])
+
+    self.vm.inputs.saveButtonTapped()
+    self.activityIndicatorShouldShow.assertValues([true])
+
+    let error = NSError(domain: "deadbeef", code: 1, userInfo: nil)
+
+    self.vm.inputs.stripeError(error)
+    self.scheduler.advance()
+
+    self.addNewCardFailure.assertValues([error.localizedDescription])
+    self.activityIndicatorShouldShow.assertValues([true, false])
+  }
+
+  func testAddCardFailure_GraphError() {
+    let error = GraphError.emptyResponse(nil)
+
+    withEnvironment(apiService: MockService(addNewCreditCardError: error)) {
+      self.vm.inputs.viewDidLoad()
+      self.vm.inputs.cardholderNameChanged("Native Squad")
+      self.vm.inputs.cardholderNameTextFieldReturn()
+      self.paymentDetailsBecomeFirstResponder
+        .assertValueCount(1, "First responder after editing cardholder name.")
+
+      self.vm.inputs.creditCardChanged(cardNumber: "4242 4242 4242 4242",
+                                        expMonth: 11,
+                                        expYear: 99,
+                                        cvc: "123")
+      self.vm.inputs.paymentInfo(valid: true)
+      self.saveButtonIsEnabled.assertValues([true])
+      self.vm.inputs.saveButtonTapped()
+      self.activityIndicatorShouldShow.assertValues([true])
+      self.vm.inputs.stripeCreated("stripe_deadbeef", stripeID: "stripe_deadbeefID")
+
+      self.scheduler.advance()
+
+      self.addNewCardFailure.assertValues([error.localizedDescription])
+      self.activityIndicatorShouldShow.assertValues([true, false])
+
+    }
+  }
+
+  func testBecomeFirstResponderAndSaveEnabled() {
```

Comment:
I think this test could be broken up in 2 pieces: one for `becomeFirstResponder` part and the other for `saveEnabled`